library/src/main/assets/leaflet_map.html

```diff
@@ -0,0 +1,502 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <meta name="viewport" content="initial-scale=1.0, user-scalable=no">
+    <meta charset="utf-8" />
+
+    <link rel="stylesheet" href="file:///android_asset/leaflet_1.3.1/leaflet.css" />
+    <script src="file:///android_asset/leaflet_1.3.1/leaflet.js" type="text/javascript"></script>
+    <script src="file:///android_asset/proj4leaflet.min.js" type="text/javascript"></script>
+    <style type="text/css">
+    body {
+      padding: 0;
+      margin: 0;
+    }
+    html, body, #map {
+      width: 100%;
+      height: 100%;
+    }
+    </style>
+</head>
+<body>
+
+<div id='map'></div>
+
+<script>
+var x_PI = Math.PI * 3000.0 / 180.0;
+function GCJ02toBD09(latlng) {
+  var lat, lng;
+  if (Array.isArray(latlng)) {
+    lat = latlng[0], lng = latlng[1];
+  } else {
+    lat = latlng.lat, lng = latlng.lng;
+  }
+  var z = Math.sqrt(lng * lng + lat * lat) + 0.00002 * Math.sin(lat * x_PI);
+  var theta = Math.atan2(lat, lng) + 0.000003 * Math.cos(lng * x_PI);
+  var bd_lng = z * Math.cos(theta) + 0.0065;
+  var bd_lat = z * Math.sin(theta) + 0.006;
+  if (Array.isArray(latlng)) {
+    return [bd_lat, bd_lng];
+  } else {
+    return {'lat': bd_lat, 'lng': bd_lng};
+  }
+}
+
+function BD09toGCJ02(latlng) {
+  var lat, lng;
+  if (Array.isArray(latlng)) {
+    lat = latlng[0], lng = latlng[1];
+  } else {
+    lat = latlng.lat, lng = latlng.lng;
+  }
+  var x = lng - 0.0065;
+  var y = lat - 0.006;
+  var z = Math.sqrt(x * x + y * y) - 0.00002 * Math.sin(y * x_PI);
+  var theta = Math.atan2(y, x) - 0.000003 * Math.cos(x * x_PI);
+  var gg_lng = z * Math.cos(theta);
+  var gg_lat = z * Math.sin(theta);
+  if (Array.isArray(latlng)) {
+    return [gg_lat, gg_lng];
+  } else {
+    return {'lat': gg_lat, 'lng': gg_lng};
+  }
+}
+
+function convertLatlng(latlng) {
+  if (isBaiduMap()) return GCJ02toBD09(latlng);
+  return latlng;
+}
+
+function reverseConvertLatlng(latlng) {
+  if (isBaiduMap()) return BD09toGCJ02(latlng);
+  return latlng;
+}
+
+function convertLatlngs(latlngs) {
+  len = latlngs.length;
+  var convertedLatlngs = new Array(len);
+  for (var i = 0; i < len; i++) {
+    convertedLatlngs[i] = convertLatlng(latlngs[i]);
+  }
+  return convertedLatlngs;
+}
+
+function isBaiduMap() {
+  return 'MAPURL'=='Baidu';
```

Comment:
`'MAPURL'=='Baidu'` my JS skills area bit rusty but I don't think this will ever be truthy

Reply:
There's a logic that we'll replace the `MAPURL` string by `mapType` string. But it's very tricky, so I'll find another way to do this.